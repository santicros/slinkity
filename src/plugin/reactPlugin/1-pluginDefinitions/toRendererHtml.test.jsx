const { toRendererHtml } = require('./toRendererHtml')
const React = require('react')

describe('toRendererHtml', () => {
  describe('render: static', () => {
    it('should generate HTML without a mount point', () => {
      const Component = () => (
        <>
          <h2>This test file is:</h2>
          <ul>
            <li>Very cool</li>
            <li>Extremely helpful</li>
          </ul>
        </>
      )

      expect(toRendererHtml({ Component, render: 'static' })).toMatchSnapshot()
    })
  })
  describe('render: eager or lazy', () => {
    it('should wrap HTML in mount point', () => {
      const Component = () => (
        <section>
          <h2>Our eager hydration</h2>
          <p>Is extremely useful</p>
          <p>
            Learn more about it <a href="https://slinkity.dev/docs/partial-hydration/">here</a>
          </p>
        </section>
      )

      expect(toRendererHtml({ Component, render: 'eager', id: 27 })).toMatchSnapshot()
    })
    it('should apply props to the component', () => {
      const Component = ({ logo, links }) => (
        <nav>
          <img src={logo.src} alt={logo.alt} />
          {links.map(({ href, label }) => (
            <a key={href} href={href}>
              {label}
            </a>
          ))}
        </nav>
      )
      const props = {
        logo: {
          src: 'accessible-logo.png',
          alt: 'Great logo',
        },
        links: [
          {
            href: '/home',
            label: 'Home',
          },
          {
            href: '/about',
            label: 'About',
          },
          {
            href: '/contact',
            label: 'Contact',
          },
        ],
      }

      expect(toRendererHtml({ Component, render: 'lazy', id: 62, props })).toMatchSnapshot()
    })
  })
})
