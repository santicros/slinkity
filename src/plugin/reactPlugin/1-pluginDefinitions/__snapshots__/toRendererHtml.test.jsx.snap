// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toRendererHtml render: eager or lazy should apply props to the component 1`] = `"<slinkity-react-mount-point data-s-id=\\"62\\"><nav data-reactroot=\\"\\"><img src=\\"accessible-logo.png\\" alt=\\"Great logo\\"/><a href=\\"/home\\">Home</a><a href=\\"/about\\">About</a><a href=\\"/contact\\">Contact</a></nav></slinkity-react-mount-point>"`;

exports[`toRendererHtml render: eager or lazy should wrap HTML in mount point 1`] = `"<slinkity-react-mount-point data-s-id=\\"27\\"><section data-reactroot=\\"\\"><h2>Our eager hydration</h2><p>Is extremely useful</p><p>Learn more about it <a href=\\"https://slinkity.dev/docs/partial-hydration/\\">here</a></p></section></slinkity-react-mount-point>"`;

exports[`toRendererHtml render: static should generate HTML without a mount point 1`] = `"<h2>This test file is:</h2><ul><li>Very cool</li><li>Extremely helpful</li></ul>"`;
